src/validation.cpp

```diff
@@ -1819,6 +1819,8 @@ bool CChainState::ConnectBlock(const CBlock& block, CValidationState& state, CBl
     CBlockIndex *pindexBIP34height = pindex->pprev->GetAncestor(chainparams.GetConsensus().BIP34Height);
     //Only continue to enforce if we're below BIP34 activation height or the block hash at that height doesn't correspond.
     fEnforceBIP30 = fEnforceBIP30 && (!pindexBIP34height || !(pindexBIP34height->GetBlockHash() == chainparams.GetConsensus().BIP34Hash));
+    // And we're not assuming the block is valid.
+    fEnforceBIP30 = fEnforceBIP30 && !fAssumeValid;
```

Comment:
Yes, but in the exact same way that an assumevalid chain with invalid scripts would cause that today.

I don't see a incentives issue, as violating bip30 really only allows you to destroy coins.

Am I missing something?

Reply:
Currently, if a bad assumevalid hash was given, we could still reorg to the honest chain if it had more work and we found a peer advertising it. The way this code is written would preclude that if an assumevalid hash was given for a chain where bip 30 was violated, which can prevent us from correctly reorging.

Reply:
From IRC:

> [08:29] <sdaftuar> Anyway this would be a simple fix to your PR — just ensure that you only skip the bip30 checks if assume valid is set and the assume valid block hash builds on the known bip34 activation block hash; that would ensure that we only skip the bip30 checks on blocks we know to be safe from this issue